### PR TITLE
New profile events for parallel replicas

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 908c7d78d5faf5cf00c3551ba8e7868cfa15017c with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69706/908c7d78d5faf5cf00c3551ba8e7868cfa15017c/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0369aaea87cdf656357fb81d3e1a9fdbec42986d/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Storages/MergeTree/MergeTreeSelectProcessor.cpp [hunk 1]
```diff
@@ -15,6 +15,39 @@
...
+    TelemetryWrapper(Func callback_, ProfileEvents::Event event_, std::string span_name_)
+        : callback(std::move(callback_)), event(event_), span_name(std::move(span_name_))
+    {
+    }
+
+    template <typename... Args>
+    auto operator()(Args &&... args)
+    {
+        DB::OpenTelemetry::SpanHolder span(span_name);
+        DB::ProfileEventTimeIncrement<DB::Time::Microseconds> increment(event);
```
Comment (kssenii) [inline src/Storages/MergeTree/MergeTreeSelectProcessor.cpp:33]: btw we do have two ways to do the same thing: `ProfileEventTimeIncrement` and  `ProfileEvent::Timer` (looks like `auto timer = DB::CurrentThread::getProfileEvents().timer(event);`). Should we delete one of them? Just contemplating, not asking you to do anything

**Review by kssenii — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeSelectProcessor.cpp (hunk 1 above)
Comment (nickitat) [reply src/Storages/MergeTree/MergeTreeSelectProcessor.cpp:33]: never saw this one. will try to remove it

**Comment by nickitat** [comment]
during testing found #69744

**Pull request merged**